mcs/class/corlib/System.Reflection/MonoParameterInfo.cs

```diff
@@ -327,6 +313,17 @@ internal static ParameterInfo New (ParameterInfo pinfo, MemberInfo member)
 		internal static ParameterInfo New (Type type, MemberInfo member, MarshalAsAttribute marshalAs)
 		{
 			return new MonoParameterInfo (type, member, marshalAs);
-		}		
+		}
+
+		private Type[] GetCustomModifiers (bool optional) {
+			Type[] types = GetTypeModifiers (optional);
+			if (types == null)
+				return Type.EmptyTypes;
+			// Mono returns modifiers in the same order as they appear in the signatures.
+			// NetFX/CoreFX returns them in reversed order.
+			// So we just reverse the order to follow NetFX/CoreFX behavior.
+			Array.Reverse (types);
```

Comment:
We built the array in the loop so just do it there instead. Please also add tests for Field and Property custom modifiers

Reply:
Seems that custom modifiers functionality does't work properly for Field and Property. 
The following code has empty output on Mono (NetFX returns both required and optional modifiers in expected order)
```
using System;
using System.Reflection;
using System.Reflection.Emit;

public sealed class A { }
public sealed class B { }
public sealed class C { }
public sealed class D { }
class Program
{
    static void Main()
    {
        var assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("A"), AssemblyBuilderAccess.RunAndSave);
        var moduleBuilder = assemblyBuilder.DefineDynamicModule("A", "A.dll");
        var typeBuilder = moduleBuilder.DefineType("C", TypeAttributes.Class);
        var fieldBuilder = typeBuilder.DefineField(
            "F",
            typeof(string),
            new[] { typeof(A), typeof(B), typeof(A), typeof(A) },
            new[] { typeof(C), typeof(D), typeof(C), typeof(C) },
            FieldAttributes.Public);

        var type = typeBuilder.CreateType();
        var field = type.GetField("F");
        foreach (var modreq in field.GetRequiredCustomModifiers())
        {
            Console.WriteLine(modreq);
        }

        foreach (var modopt in field.GetOptionalCustomModifiers())
        {
            Console.WriteLine(modopt);
        }
    }
}
```